pkg/api/types.go

```diff
@@ -1192,13 +1195,15 @@ type ServiceSpec struct {
 	// None can be specified for headless services when proxying is not required
 	ClusterIP string `json:"clusterIP,omitempty"`
 
-	// Type determines how the service will be exposed.  Valid options: ClusterIP, NodePort, LoadBalancer
-	Type ServiceType `json:"type,omitempty"`
 
 	// ExternalIPs are used by external load balancers, or can be set by
 	// users to handle external traffic that arrives at a node.
 	ExternalIPs []string `json:"externalIPs,omitempty"`
 
+	// Only applies to Service Type: LoadBalancer
+	// LoadBalancer will get created with the IP specified in this field.
+	LoadBalancerIP string `json:"loadBalancerIP,omitempty"`
```

Comment:
(reposting comment against latest diff, so it doesn't get lost)

1. I strongly suspect this should be an opaque provider-specific string rather than implying that its an IP.

2. I feel like this is very much a provider-specific implementation detail, and should go in some provider-specific config file or some other generic key/value provider-attributes-extension mechanism.

Eg: the k8s openstack config currently requires a "subnet-id" and other monitoring options for loadbalancers.  These options are set globally via the config right now, but there are reasons you might want to vary them by service.